File: airplane_mode/shop_lease/web_form/shop_inquiry/shop_inquiry.py
import frappe
from frappe import _

def get_context(context):
	# do your magic here
	pass


# def validate(self):
# 	if self.email:
# 		self.email = self.email.strip()
# 	if self.phone:
# 		self.phone = self.phone.strip()
# 	if not self.email and not self.phone:		
# 		frappe.throw(
# 			msg=_("Please enter at least either email or phone number."),
# 			title=_("Error"),
# 		)

File: airplane_mode/shop_lease/web_form/shop_inquiry/shop_inquiry.js
frappe.ready(function () {
	// validate if phone number or email exists
	frappe.web_form.validate = () => {
		let data = frappe.web_form.get_values();
		if (data.email == "" || data.phone_number == '') {
			frappe.msgprint(_("Please provide either email or phone number."))
			return false;
		}
	};

	// set airport_shop field
	const url_params = new URLSearchParams(window.location.search);
	if (url_params.has('airport_shop')) {
		frappe.web_form.set_value('airport_shop', url_params.get('airport_shop'));
	}

	// TODO: lookup location based on user's IP and set phone country code
	$.getJSON('https://ipapi.co/json/', function (data) {
		if (data && data.country_calling_code) {
			const prefix = data.country_calling_code + "-";  // e.g. "+886"
			const field = frappe.web_form.get_field('phone_number');
			if (field && field.$wrapper) {
				field.set_value(prefix);
				// Manually trigger a change event so Frappe's internal listeners pick up the new value
				field.$input.trigger('change');
			}
		}
	}).fail(function () {
		console.log("Phone country code lookup failed. Defaulting to manual entry.");
	});

})



